### PR TITLE
Merge release into master

### PR DESCRIPTION
**Pull request by emanuelez**

### CODE REVIEW

**Comment by emanuelez** [comment]
@mc I did the merge again from scrap and fixed the problems I met on the way.

Ready for review @realm/java

File: bintrayAar.gradle [hunk 1]
```diff
@@ -0,0 +1,126 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        //classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.1.1'
```
Comment (cmelchior) [inline bintrayAar.gradle]: Remove this line?

File: bintrayAar.gradle [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    }
+}
+
+publishing {
+    publications {
+        jarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android'
+            artifact file("${buildDir}/outputs/gradle/realm-android-${currentVersion}.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayAar.gradle]: Remove out commented code

File: bintrayAar.gradle [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                root.appendNode('description', 'Realm is a mobile database: a replacement for SQLite & ORMs.')
+                root.appendNode('url', 'http://realm.io')
+                root.children().last() + commonPom
+            }
+        }
+        aarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android-library'
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayAar.gradle]: Same

File: bintrayAnnotationsProcessor.gradle [hunk 4]
```diff
@@ -0,0 +1,131 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        //classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.1.1'
```
Comment (cmelchior) [inline bintrayAnnotationsProcessor.gradle]: Remove

File: bintrayAnnotationsProcessor.gradle [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    }
+}
+
+publishing {
+    publications {
+        jarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android'
+            artifact file("${buildDir}/outputs/gradle/realm-android-${currentVersion}.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayAnnotationsProcessor.gradle]: Remove

File: bintrayAnnotationsProcessor.gradle [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+                root.appendNode('description', 'Realm is a mobile database: a replacement for SQLite & ORMs.')
+                root.appendNode('url', 'http://realm.io')
+                root.children().last() + commonPom
+            }
+        }
+        aarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android-library'
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayAnnotationsProcessor.gradle]: Remove

File: bintrayGradlePlugin.gradle [hunk 7]
```diff
@@ -0,0 +1,131 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        //classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.1.1'
```
Comment (cmelchior) [inline bintrayGradlePlugin.gradle]: Remove

File: bintrayGradlePlugin.gradle [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    }
+}
+
+publishing {
+    publications {
+        jarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android'
+            artifact file("${buildDir}/outputs/gradle/realm-android-${currentVersion}.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayGradlePlugin.gradle]: Remove

File: bintrayGradlePlugin.gradle [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                root.appendNode('description', 'Realm is a mobile database: a replacement for SQLite & ORMs.')
+                root.appendNode('url', 'http://realm.io')
+                root.children().last() + commonPom
+            }
+        }
+        aarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android-library'
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayGradlePlugin.gradle]: Remove

File: bintrayJar.gradle [hunk 10]
```diff
@@ -0,0 +1,126 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        //classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.1.1'
```
Comment (cmelchior) [inline bintrayJar.gradle]: Remove

File: bintrayJar.gradle [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    }
+}
+
+publishing {
+    publications {
+        jarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android'
+            artifact file("${buildDir}/outputs/gradle/realm-android-${currentVersion}.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayJar.gradle]: Remove

File: bintrayJar.gradle [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+                root.appendNode('description', 'Realm is a mobile database: a replacement for SQLite & ORMs.')
+                root.appendNode('url', 'http://realm.io')
+                root.children().last() + commonPom
+            }
+        }
+        aarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android-library'
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline bintrayJar.gradle]: Remove

File: build.gradle [hunk 13]
```diff
@@ -296,8 +300,8 @@ publishing {
             groupId 'io.realm'
             artifactId = 'realm-android'
             artifact file("${buildDir}/outputs/gradle/realm-android-${currentVersion}.jar")
-            artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
-            artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-javadoc.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline build.gradle]: Remove

File: build.gradle [hunk 14]
```diff
@@ -395,8 +399,7 @@ bintray {
 }
 
 release {
-  dependsOn distributionPackage
-  push = true
+  push = false
```
Comment (cmelchior) [inline build.gradle]: How does this work if executed on the build server? Then the change would only be locally there?

File: build.gradle [hunk 15]
```diff
@@ -310,8 +314,8 @@ publishing {
             groupId 'io.realm'
             artifactId = 'realm-android-library'
             artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
-            artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
-            artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-javadoc.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (cmelchior) [inline build.gradle]: Remove

**Comment by cmelchior** [comment]
Duplicating the publication info across 5 files looks extremely difficult to maintain, but if it is a temporary solution I guess it is ok.

File: build.gradle (hunk 14 above)
Comment (emanuelez) [inline build.gradle]: We're not ready for the build server yet. In a few iterations this will change.

File: bintrayAar.gradle (hunk 2 above)
Comment (emanuelez) [inline bintrayAar.gradle]: I'm keeping those as a reminder to find a way to import them again.

File: bintrayAar.gradle [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+        }
+    }
+}
+
+bintray {
+    user = project.hasProperty('bintrayUser') ? bintrayUser : 'noUser'
+    key = project.hasProperty('bintrayKey') ? bintrayKey : 'noKey'
+
+    dryRun = false
+    publish = false
```
Comment (cmelchior) [inline bintrayAar.gradle]: Do we want to manually publish all of these?

File: bintrayAnnotations.gradle [hunk 17]
```diff
@@ -0,0 +1,126 @@
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        //classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.1.1'
```
Comment (kneth) [inline bintrayAnnotations.gradle]: Can it be removed?

File: bintrayAnnotations.gradle [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+    }
+}
+
+publishing {
+    publications {
+        jarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android'
+            artifact file("${buildDir}/outputs/gradle/realm-android-${currentVersion}.jar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (kneth) [inline bintrayAnnotations.gradle]: Remove?

File: bintrayAnnotations.gradle [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+                root.appendNode('description', 'Realm is a mobile database: a replacement for SQLite & ORMs.')
+                root.appendNode('url', 'http://realm.io')
+                root.children().last() + commonPom
+            }
+        }
+        aarLibrary(MavenPublication) {
+            groupId 'io.realm'
+            artifactId = 'realm-android-library'
+            artifact file("${rootDir}/realm/realm-library/build/outputs/aar/realm-library-release.aar")
+            //artifact file("${rootDir}/realm/realm-library/build/libs/realm-android-${currentVersion}-sources.jar")
```
Comment (kneth) [inline bintrayAnnotations.gradle]: Remove?

File: bintrayAar.gradle (hunk 16 above)
Comment (emanuelez) [inline bintrayAar.gradle]: For the moment, yes. When automation kick in, not anymore.

File: version.txt [hunk 20]
```diff
@@ -1 +1,2 @@
-0.86.0-SNAPSHOT
+0.85.2-SNAPSHOT
```
Comment (beeender) [inline version.txt]: Shouldn't this be `0.86.0-SNAPSHOT` ?

Comment (emanuelez) [inline version.txt]: No, the release plugin now takes care of the version file. We don't need to touch it anymore :)

Comment (dalinaum) [inline version.txt]: If the relase plugin takes care of ther version, can we fill up this the blank? I think `0.85.2-SNAPSHOT` is little confusing.

**Comment by cmelchior** [comment]
:+1:

File: version.txt (hunk 20 above)
Comment (emanuelez) [inline version.txt]: I'm not sure what you mean @dalinaum

**Pull request merged**